### PR TITLE
Allowing users to be able to view operations logs as cluster-reader i…

### LINKED ISSUE #42: Allow Cluster Reader to have access to the .all alias and .operations index
We should allow users the ability to configure their plugin to determine if just `cluster-admin` or `cluster-reader` as well, can see the infrastructure logs.

### PR DESCRIPTION
**Pull request by ewolinetz**
…f configured

Resolves #42 

@jcantrill @richm PTAL

### CODE REVIEW

**Comment by richm** [comment]
lgtm but I don't know the openshift api

**Comment by jcantrill** [comment]
LGTM assuming 'tokenreview' is enough to indicate a user is a clusterreader.

**Comment by jcantrill** [comment]
After reading @liggit response, apparently we should revisit how we determine if a user is a cluster-admin since we are trying to 'educate dolphins'  ...which would probably be pretty easy since they are smart animals

**Comment by ewolinetz** [comment]
@jcantrill updated pattern for checking roles, we will need to update our deployer to let the ES serviceaccount be able to 'get' 'clusterrolebindings'.  `oadm policy add-cluster-role-to-user {the-new-role} {es-serviceaccount-user}`

PTAL

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/SearchGuardRolesMapping.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ public void syncFrom(UserProjectCache cache, final String userProfilePrefix){
 				builder.addUser(projectRoleName, username);
 			}
 			
-			if ( cache.isClusterAdmin(username) ) {
+			if ( cache.isOperationsUser(username) ) {
```
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/SearchGuardRolesMapping.java]: For our needs, is an operations user who could be a cluster-admin or cluster-reader the same 'ADMIN_ROLE'?  Do we have concerns that a cluster-reader may have access to logs they should not?

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java [hunk 2]
```diff
@@ -270,25 +267,35 @@ private boolean isBlacklistProject(String project) {
 		return ArrayUtils.contains(operationsProjects, project.toLowerCase());
 	}
 
-	private boolean isClusterAdmin(final String token){
-		ConfigBuilder builder = new ConfigBuilder()
-				.withOauthToken(token);
+	private boolean isOperationsUser(final String username){
+		ConfigBuilder builder = new ConfigBuilder();
+		OpenShiftClient osClient = new DefaultOpenShiftClient(builder.build());
```
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: I think this class is 'Closeable' which means you can:
```
try(OpenShiftClient osClient = new DefaultOpenShiftClient(builder.build())){
}catch(Exception){}
```
and not even need to explicit close the client

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/UserProjectCache.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 	 * @param project  the project
 	 * @param binding  the binding to get user info
 	 */
-	void update(final String user, Set<String> projects, boolean clusterAdmin);
+	void update(final String user, Set<String> projects, boolean operationsUser);
```
Comment (jcantrill) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/UserProjectCache.java]: Please update the param comments

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java (hunk 2 above)
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java]: it flags it if we explicitly don't close it... i'll wrap it though

**Comment by ewolinetz** [comment]
@jcantrill addressed comments

**Comment by jcantrill** [comment]
LGTM

**Comment by ewolinetz** [comment]
[merge]

**Comment by ewolinetz** [comment]
@jcantrill can we do a release with this please?

**Comment by jcantrill** [comment]
released as part of 2.3.5.7

**Pull request merged**